### PR TITLE
satellite/console: forgot password, resend email endpoints added, default http route replaced with gorilla mux

### PR DESCRIPTION
**Pull request by Qweder93**
What:  forgot password, resend email endpoints added, default http route replaced with gorilla mux

Why:

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/console/consoleweb/server.go [hunk 1]
```diff
@@ -114,27 +115,36 @@ func NewServer(logger *zap.Logger, config Config, service *console.Service, mail
...
 	router.Handle("/api/v0/graphql", http.HandlerFunc(server.grapqlHandler))
-	router.Handle("/api/v0/token", http.HandlerFunc(server.tokenHandler))
-	router.Handle("/api/v0/register", http.HandlerFunc(server.registerHandler))
 	router.Handle("/registrationToken/", http.HandlerFunc(server.createRegistrationTokenHandler))
 	router.Handle("/robots.txt", http.HandlerFunc(server.seoHandler))
 
+	authRouter := router.PathPrefix("/api/auth").Subrouter()
+	authRouter.Use(server.authMiddlewareHandler)
+
+	authRouter.Handle("/token", http.HandlerFunc(authController.Token)).Methods("POST")
```
Comment (crawter) [inline satellite/console/consoleweb/server.go]: authRouter.HandleFunc("/token", authController.Token).Methods("POST")

File: satellite/console/consoleweb/server.go [hunk 2]
```diff
@@ -114,27 +115,36 @@ func NewServer(logger *zap.Logger, config Config, service *console.Service, mail
...
+	router.Handle("/api/v0/payments/account", http.HandlerFunc(paymentController.SetupAccount))
+
 	if server.config.StaticDir != "" {
 		router.Handle("/activation/", http.HandlerFunc(server.accountActivationHandler))
 		router.Handle("/password-recovery/", http.HandlerFunc(server.passwordRecoveryHandler))
 		router.Handle("/cancel-password-recovery/", http.HandlerFunc(server.cancelPasswordRecoveryHandler))
 		router.Handle("/usage-report/", http.HandlerFunc(server.bucketUsageReportHandler))
-		router.Handle("/static/", server.gzipHandler(http.StripPrefix("/static", fs)))
-		router.Handle("/", http.HandlerFunc(server.appHandler))
+		router.PathPrefix("/static/").Handler(http.StripPrefix("/static", fs))
```
Comment (crawter) [inline satellite/console/consoleweb/server.go]: router.PathPrefix("/static/").Handler(server.gzipHandler(http.StripPrefix("/static", fs)))

File: satellite/console/consoleweb/consoleapi/auth.go [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+package consoleapi
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
```
Comment (crawter) [inline satellite/console/consoleweb/consoleapi/auth.go]: reorder

File: satellite/console/consoleweb/consoleapi/auth.go [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+		service:               service,
+		mailService:           mailService,
+		ExternalAddress:       externalAddress,
+		LetUsKnowURL:          letUsKnowURL,
+		TermsAndConditionsURL: termsAndConditionsURL,
+		ContactInfoURL:        contactInfoURL,
+	}
+}
+
+// Token authenticates User by credentials and returns auth token.
```
Comment (crawter) [inline satellite/console/consoleweb/consoleapi/auth.go]: user

File: satellite/console/consoleweb/consoleapi/auth.go [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"github.com/zeebo/errs"
```
Comment (crawter) [inline satellite/console/consoleweb/consoleapi/auth.go]: reorder

File: satellite/console/consoleweb/consoleapi/auth.go [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/post"
+	"storj.io/storj/pkg/auth"
+	"storj.io/storj/satellite/console"
+	"storj.io/storj/satellite/console/consoleweb/consoleql"
+	"storj.io/storj/satellite/mailservice"
+)
+
+var Error = errs.Class("satellite console auth api error")
```
Comment (crawter) [inline satellite/console/consoleweb/consoleapi/auth.go]: we use named logger, so lets leave only "console auth api error" not to duplicate text

File: satellite/console/consoleweb/server.go [hunk 7]
```diff
@@ -114,27 +114,35 @@ func NewServer(logger *zap.Logger, config Config, service *console.Service, mail
...
-	router.Handle("/api/v0/token", http.HandlerFunc(server.tokenHandler))
-	router.Handle("/api/v0/register", http.HandlerFunc(server.registerHandler))
-	router.Handle("/registrationToken/", http.HandlerFunc(server.createRegistrationTokenHandler))
-	router.Handle("/robots.txt", http.HandlerFunc(server.seoHandler))
+	router.HandleFunc("/api/v0/graphql", server.grapqlHandler)
+	router.HandleFunc("/registrationToken/", server.createRegistrationTokenHandler)
+	router.HandleFunc("/robots.txt", server.seoHandler)
+
+	authRouter := router.PathPrefix("/api/auth").Subrouter()
+	authRouter.Use(server.authMiddlewareHandler)
```
Comment (crawter) [inline satellite/console/consoleweb/server.go]: does it mean that we authorize twice?

File: satellite/console/consoleweb/consoleapi/auth.go [hunk 8]
```diff
@@ -155,6 +160,109 @@ func (a *Auth) PasswordChange(w http.ResponseWriter, r *http.Request) {
...
+// ForgotPassword creates password-reset token and sends email to user.
+func (a *Auth) ForgotPassword(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+	params := mux.Vars(r)
+	email, ok := params["email"]
+	if !ok {
+		err = errs.New("email expected")
+
```
Comment (crawter) [inline satellite/console/consoleweb/consoleapi/auth.go]: this empty line looks weird

File: satellite/console/consoleweb/consoleapi/auth.go [hunk 9]
```diff
@@ -155,6 +160,109 @@ func (a *Auth) PasswordChange(w http.ResponseWriter, r *http.Request) {
...
+		userName = user.FullName
+	}
+
+	contactInfoURL := a.ContactInfoURL
+	termsAndConditionsURL := a.TermsAndConditionsURL
+
+	a.mailService.SendRenderedAsync(
+		ctx,
+		[]post.Address{{Address: user.Email, Name: userName}},
+		&consoleql.AccountActivationEmail{
```
Comment (rikysya) [inline satellite/console/consoleweb/consoleapi/auth.go]: we can now move this out from `consoleql` pkg, to `consoleapi`

**Pull request merged**